exercises/container-security/docker-hub-cli/test/getDigest.test.js

```diff
@@ -0,0 +1,47 @@
+const { validateDigestArgs, getDigest } = require('../cli/index');
+const { getDockerContentDigest } = require('../lib/index.js');
+
+describe('validate args', () => {
+  test('validate valid args', () => {
+    const args = { repo: 'alpine', tag: '3' };
+    expect(validateDigestArgs(args)).toBeTruthy();
+  });
+  test('validate missing arg', () => {
+    const args = { repo: 'alpine' };
+    expect(validateDigestArgs(args)).toBeFalsy();
+  });
+  test('validate wrong type arg', () => {
+    const args = { repo: 1, tag: true };
+    expect(validateDigestArgs(args)).toBeFalsy();
+  });
+});
+
+jest.mock('../lib/index.js', () => ({
+  getDockerContentDigest: jest.fn()
+}))
+
+describe('test getDigest', () => {
+    test('valid repo and tag', async () => {
+    getDockerContentDigest.mockReturnValueOnce('sha256:abcd');
+    const args = { repo: 'alpine', tag: '3' };
+    expect(await getDigest(args.repo, args.tag)).toEqual('sha256:abcd');
+  });
+  test('handles unknown error', async () => {
+    getDockerContentDigest.mockRejectedValue(new Error());
+    const args = { repo: 'alpine', tag: '3'};
+    await expect(getDigest(args.repo, args.tag)).rejects.toThrow();
+  });
+  test('handles unauthorized Error for non existent tag', async () => {
+    const unAuthorizedError = { statusCode: 404 };
+    getDockerContentDigest.mockRejectedValue(unAuthorizedError);
+    const args = { repo: 'alpine', tag: 'nonExistentTag'};
+    await !expect(getDigest(args.repo, args.tag)).rejects.toThrow();
+  });
+  test('handles not found Error for non existent', async () => {
+    const notFoundError = { statusCode: 401 };
+    getDockerContentDigest.mockRejectedValue(notFoundError);
+    const args = { repo: 'nonExistentRepo', tag: '3'};
+    await !expect(getDigest(args.repo, args.tag)).rejects.toThrow();
+  });
+});
+
```

Comment:
Its considered good practice to have the source and test files symmetrically named, e.g.: `src/cli/index.js` and `test/cli/index.test.js`. This way its really easy to find tests and also obvious what belongs in the test (only the public interface of the source file) and what not.

You could also start on the other end in the source, and first pull `{ validateDigestArgs, getDigest }` into `src/lib/get-digest.js`, and then test file make more sense `test/lib/get-digest.test.js`...

p.s. I know we don't really follow this convention internally 🤷‍♂

Reply:
I see what you mean, I think I'll add the extra dir level in the test dir.

Reply:
I'll add the additional dir under the test dir. The Goal is to make things as simple as possible so I'll add it if you think it helps with visibility and navigation.